providers/org/datacite/normalizer.py

```diff
@@ -28,243 +74,402 @@ def force_text(data):
 
+        contributor_type = tools.Try(ctx.contributorType)
 
-class ThroughLinks(Parser):
-    link = tools.Delegate(Link, ctx)
+        # v.4 new givenName and familyName properties
+        given_name = tools.OneOf(
+            ctx.creatorName['@givenName'],
+            ctx.contributorName['@givenName'],
+            tools.Static(None)
+        )
+        family_name = tools.OneOf(
+            ctx.creatorName['@familyName'],
+            ctx.contributorName['@familyName'],
+            tools.Static(None)
+        )
 
 
-class ThroughAlternateLinks(Parser):
-    schema = 'ThroughLinks'
+class FunderAgent(Parser):
+    schema = tools.OneOf(
+        tools.RunPython(
+            get_agent_type,
+            ctx
+        ),
+        tools.GuessAgentType(
+            tools.OneOf(ctx.funderName, ctx.contributorName),
+            default='organization'
+        )
+    )
 
-    link = tools.Delegate(AlternateLink, ctx)
+    name = tools.OneOf(ctx.funderName, ctx.contributorName)
 
+    identifiers = tools.Map(
+        tools.Delegate(AgentIdentifier),
+        tools.OneOf(
+            ctx.funderIdentifier,
+            tools.RunPython(
+                force_text,
+                ctx.nameIdentifier
+            ),
+            tools.Static(None)
+        )
+    )
 
-class ThroughRelatedLinks(Parser):
-    schema = 'ThroughLinks'
+    class Extra:
+        name_identifier = tools.Try(ctx.nameIdentifier)
+        name_identifier_scheme = tools.Try(ctx.nameIdentifier['@nameIdentifierScheme'])
+        name_identifier_scheme_uri = tools.Try(ctx.nameIdentifier['@schemeURI'])
 
-    link = tools.Delegate(RelatedLink, ctx)
+        funder_identifier = ctx.funderIdentifier
+        funder_identifier_type = ctx.funderIdentifierType
 
+        contributor_type = tools.Try(ctx.contributorType)
 
-class Affiliation(Parser):
-    pass
 
+class HostAgent(Parser):
+    schema = tools.GuessAgentType(ctx, default='organization')
 
-class ContributorInstitution(Parser):
-    schema = 'Institution'
+    name = tools.Try(ctx.contributorName)
 
-    name = ctx.contributorName
+    identifiers = tools.Map(
+        tools.Delegate(AgentIdentifier),
+        tools.Try(
+            tools.RunPython(
+                force_text,
+                ctx.nameIdentifier
+            )
+        )
+    )
 
     class Extra:
+        name_identifier = tools.Try(ctx.nameIdentifier)
+        name_identifier_scheme = tools.Try(ctx.nameIdentifier['@nameIdentifierScheme'])
+        name_identifier_scheme_uri = tools.Try(ctx.nameIdentifier['@schemeURI'])
+
         contributor_type = tools.Try(ctx.contributorType)
 
 
-class ContributorOrganization(Parser):
-    schema = 'Organization'
+class PublisherAgent(Parser):
+    schema = tools.GuessAgentType(ctx, default='organization')
 
-    name = ctx.contributorName
+    name = ctx
 
     class Extra:
         contributor_type = tools.Try(ctx.contributorType)
```

Comment:
`ctx` is the publisher name here, so it'll always be a string, and this `Try` emits a warning for every record. I think you can get rid of it.